src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,40 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+"""
+    transpose(df::AbstractDataFrame, src_namescol=1, dest_namescol=names(df)[src_namescol]; copycols::Bool=false, makeunique=false, promote_type::Bool=true)
```

Comment:
please make sure to obey 80 characters per line as a soft limit and 92 as a hard limit